functions/safe_filename.pp

```diff
@@ -0,0 +1,70 @@
+# @summary Returns a string that is safe for firewalld filenames
```

Comment:
Do you consider this private ie should it also have `@api private`?

Reply:
@alexjfisher No, this is not private. In fact, I need it somewhere else because not everything in here is safe yet.